### PR TITLE
Add payment microservice.

### PR DESCRIPTION
**Pull request by maksim-luzin**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,40 +1,37 @@
...
+        "install:backend": "cd backend && npm install",
+        "install:frontend": "cd frontend && npm install",
+        "install:payment": "cd payment && npm install",
+        "install:all": "npm install && npm run build:shared && npm run install:gateway && npm run install:backend && npm run install:frontend, && npm run install:payment",
+        "build:shared": "cd shared && npm install && npm run build",
+        "start:gateway": "cd gateway && npm run start",
+        "start:backend": "cd backend && npm run start",
+        "start:frontend": "cd frontend && npm run start",
+        "start:payment": "cd payment && npm run start",
+        "start": "concurrently -k -p \"{name}: {time}\" -n \"FRONTEND,GATEWAY,BACKEND\" \"npm:start:frontend\" \"npm:start:gateway\" \"npm:start:backend\" \"npm:start:payment\""
```
Comment (mtseluiko) [inline package.json]: also add PAYMENT title

File: package.json [hunk 2]
```diff
@@ -1,40 +1,37 @@
...
-    "hooks": {
-      "pre-commit": "lint-staged"
-    }
-  },
-  "lint-staged": {
-    "**/*.(js|ts|tsx)": [
-      "npm run lint"
-    ]
-  },
+    "scripts": {
```
Comment (mtseluiko) [inline package.json]: use 2 spaces for tabs

File: payment/src/api/routes/index.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import MicroMq from 'micromq';
+import { Path } from '../../common/enums';
+import paymentRouter from './payment';
+import services from '../../services';
+
+const initRoutes = (app: MicroMq) => paymentRouter(services, Path.Payment, app);
```
Comment (mtseluiko) [inline payment/src/api/routes/index.ts]: The "backend" microservice routing have changed. Please update here, as well.

File: payment/src/data/repositories/payment.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+export class PaymentRepository {
```
Comment (mtseluiko) [inline payment/src/data/repositories/payment.ts]: Please, connect it also to database like in the Backend microservice

**Review by mtseluiko — CHANGES_REQUESTED** [review]

File: payment/src/server.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import MicroMq from 'micromq';
+import { Project } from 'hypecrafter-shared';
+import initRoutes from './api/routes';
+import { env } from './env';
+
+const { rabbit } = env.app;
```
Comment (mtseluiko) [inline payment/src/server.ts:8]: connect this microservice to the gateway

File: package.json (hunk 1 above)
Comment (maksim-luzin) [reply package.json]: I added

File: package.json (hunk 2 above)
Comment (maksim-luzin) [reply package.json]: I rebuilt.

File: package.json (hunk 1 above)
Comment (mtseluiko) [reply package.json]: Here: -n \"FRONTEND,GATEWAY,BACKEND\"

File: package.json (hunk 2 above)
Comment (maksim-luzin) [reply package.json]: I corrected.

File: payment/src/api/routes/index.ts (hunk 3 above)
Comment (maksim-luzin) [reply payment/src/api/routes/index.ts]: I changed.

File: payment/src/data/repositories/payment.ts (hunk 4 above)
Comment (maksim-luzin) [reply payment/src/data/repositories/payment.ts]: I examined the table and consulted with your colleague. Changed a little. It is now an associative donate table. It connects the user profile to projects.

File: payment/src/server.ts (hunk 5 above)
Comment (maksim-luzin) [reply payment/src/server.ts:8]: I changed.

File: gateway/package.json [hunk 6]
```diff
@@ -6,12 +6,7 @@
   "scripts": {
     "start": "cross-env NODE_ENV=development nodemon -r ts-node --files ./src/server.ts",
     "lint": "eslint . --ext .ts",
-    "lint:write": "eslint . --ext .ts --fix",
-    "add-migration": "ts-node node_modules/typeorm/cli.js migration:generate -n",
-    "migrate": "ts-node node_modules/typeorm/cli.js migration:run",
-    "migration-revert": "ts-node node_modules/typeorm/cli.js migration:revert",
-    "seed": "ts-node ./src/data/seeders",
-    "typeorm": "ts-node node_modules/typeorm/cli.js"
+    "lint:write": "eslint . --ext .ts --fix"
```
Comment (mtseluiko) [inline gateway/package.json]: why removed this?

Comment (maksim-luzin) [reply gateway/package.json]: I could create a migration when all models are in one place. Models in different microservices do not create migration.

File: package.json [hunk 7]
```diff
@@ -3,37 +3,34 @@
...
-    "seed:all": "npm run seed:backend && npm run seed:gateway",
-    "start": "concurrently -k -p \"{name}: {time}\" -n \"FRONTEND,GATEWAY,BACKEND\" \"npm:start:frontend\" \"npm:start:gateway\" \"npm:start:backend\""
+    "start:payment": "cd payment && npm run start",
+    "start": "concurrently -k -p \"{name}: {time}\" -n \"FRONTEND,GATEWAY,BACKEND,PAYMENT\" \"npm:start:frontend\" \"npm:start:gateway\" \"npm:start:backend\" \"npm:start:payment\""
   },
   "devDependencies": {
-    "concurrently": "^6.2.0",
-    "husky": "^4.3.8",
-    "lint-staged": "^11.1.1"
+      "concurrently": "^6.2.0",
```
Comment (mtseluiko) [inline package.json]: spaces again. Please, check setups of autoformatting in your editor if it adds it automatically

File: gateway/package.json (hunk 6 above)
Comment (mtseluiko) [reply gateway/package.json]: microservices should be independent. Maybe we'll even use separate databases

File: payment/src/data/entities/donate.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { Entity, Column, ManyToOne } from 'typeorm';
+import { AbstractEntity } from './abstract';
+import { UserProfile } from '../../../../backend/src/data/entities/userProfile';
```
Comment (mtseluiko) [inline payment/src/data/entities/donate.ts]: Please, remove this relation. Microservices should be independent. We'll store all payment transactions in the Payment database, and userId will be passed from the gateway/backend.

**Review by mtseluiko — CHANGES_REQUESTED** [review]

File: gateway/package.json (hunk 6 above)
Comment (maksim-luzin) [reply gateway/package.json]: Good. Which table from the schema to use?

File: package.json (hunk 7 above)
Comment (maksim-luzin) [reply package.json]: Ok

File: payment/src/data/entities/donate.ts (hunk 8 above)
Comment (maksim-luzin) [reply payment/src/data/entities/donate.ts]: Good. Which table from the schema to use?

File: package.json [hunk 9]
```diff
@@ -1,45 +1,42 @@
...
+    "install:notification": "cd notification && npm install",
+    "install:payment": "cd payment && npm install",
+    "install:all": "npm install && npm run build:shared && npm run install:gateway && npm run install:backend && npm run install:frontend, && npm run install:payment",
+    "build:shared": "cd shared && npm install && npm run build",
+    "start:gateway": "cd gateway && npm run start",
+    "start:backend": "cd backend && npm run start",
+    "start:frontend": "cd frontend && npm run start",
+    "start:notification": "cd notification && npm run start",
+    "start:payment": "cd payment && npm run start",
+    "start": "concurrently -k -p \"{name}: {time}\" -n \"FRONTEND,GATEWAY,BACKEND,NOTIFICATION,PAYMENT\" \"npm:start:frontend\" \"npm:start:gateway\" \"npm:start:backend\" \"npm:start:notification\" \"npm:start:payment\""
```
Comment (mtseluiko) [inline package.json]: it seems, here were more scripts

**Pull request merged**